src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const CopyOfStateObject = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    const currentObjectState = { ...CopyOfStateObject };
+
+    switch (action.type) {
+      case 'removeProperties':
+        for (const statesToRemove of action.keysToRemove) {
+          delete CopyOfStateObject[statesToRemove];
+          delete currentObjectState[statesToRemove];
+        }
+        result.push(currentObjectState);
+        break;
+      case 'addProperties':
+        Object.assign(currentObjectState, action.extraData);
+        Object.assign(CopyOfStateObject, action.extraData);
+        result.push(currentObjectState);
+        break;
+      case 'clear':
+        for (const keyElement in CopyOfStateObject) {
+          delete CopyOfStateObject[keyElement];
+          delete currentObjectState[keyElement];
+        }
+        result.push({});
+        break;
+      default:
+        return `some error occurred`;
+    }
```

Comment:
```suggestion
        break;
      case 'addProperties':
        Object.assign(currentObjectState, action.extraData);
        Object.assign(CopyOfStateObject, action.extraData);
        
        break;
      case 'clear':
        for (const keyElement in CopyOfStateObject) {
          delete CopyOfStateObject[keyElement];
          delete currentObjectState[keyElement];
        }
        break;
      default:
        return `some error occurred`;
    }
    
    result.push({ ...copyOfStateObject });
```
you can push one time currentObjectState. after switch case.
and you don't need currentObjectState. enough to work with copyOfStateObject and push its clone